scrapy/utils/datatypes.py

```diff
@@ -291,3 +292,29 @@ def __init__(self, seq):
 
     def __contains__(self, item):
         return item not in self.seq
+
+
+def get_item_field(item, field_name, default=None):
+    """
+    Access item fields on both "regular" and dataclass-based items
+    """
+    if is_dataclass_instance(item):
+        return getattr(item, field_name, default)
+    else:
+        return item.get(field_name, default)
+
+
+def set_item_field(item, field_name, value):
+    """
+    Set item fields on both "regular" and dataclass-based items
+    """
+    if is_dataclass_instance(item):
+        if not hasattr(item, "_field_names"):
+            from dataclasses import fields
+            item._field_names = [f.name for f in fields(item)]
```

Comment:
Is it a performance optimization, to cache available field names? I'm not sure that's good to add attributes to user's objects.

Reply:
The same applies to subscriptable_dataclass.